rdkit/Chem/AllChem.py

```diff
@@ -499,34 +542,52 @@ def GenerateStereoisomers(m,options=StereoEnumerationOptions(),verbose=False):
 
     """
     tm = Mol(m)
-    if options.onlyUnassigned:
-        possibleCenters = [x for x,y in FindMolChiralCenters(tm, force=True, includeUnassigned=True) if y=='?']
-    else:
-        possibleCenters = [x for x,y in FindMolChiralCenters(tm, force=True, includeUnassigned=True)]
-    nCenters = len(possibleCenters)
+    flippers = _getFlippers(tm, options)
+    nCenters = len(flippers)
     if not nCenters:
         yield tm
         return
-    if nCenters>options.maxNumCenters:
-        raise ValueError("nCenters (%d) larger than maxNumCenters (%d)"%(nCenters,options.maxNumCenters))
-    bitflag = (1<<nCenters)-1
-    while bitflag>=0:
-        tm = Mol(m)
+
+    # this sillyness is to deal with Python 3 getting rid of xrange
+    # and range slurping up 2**N memory in Python 2
+    range_func = range
+    if sys.version_info.major < 3:
+        range_func = xrange
+
+    if (options.maxIsomers == 0 or
+        2**nCenters <= options.maxIsomers):
+        samples = range_func(2**nCenters)
+    else:
+        if options.rand is None:
+            # deterministic random seed invariant to input atom order
+            seed = hash(tuple(sorted([(a.GetDegree(), a.GetAtomicNum()) for a in tm.GetAtoms()])))
+            rand = random.Random(seed)
+        elif isinstance(options.rand, random.Random):
+            # other implementations of Python random number generators
+            # can inherit from this class to pick up utility methods
+            rand = options.rand
+        else:
+            rand = random.Random(self.rand)
+
+        # randomly sample from the population of isomers
+        samples = rand.sample(range_func(2**nCenters), k=options.maxIsomers)
```

Comment:
This is going to die at inconvenient times:

```
>>> rand.sample(xrange(2**64), 10)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python int too large to convert to C long
```

Reply:
nCenters is max 62 with this technique

Reply:
Perhaps but in a warning, that's an absurd number of unspecified stereo anyway :)

Reply:
How much do we care about this? Python OverflowError actually seems like a reasonable outcome. Otherwise I would to need to basically re-implement random.sample using getrandbits instead. I've added a test case for the OverflowError at least.

Reply:
How about a ValueError and documenting the (current) max value?

Reply:
Sounds good, done.

Reply:
In the latest changes, this is no longer an issue as I switched to getrandbits() that can supply an arbitrary number of random bits.